torchvision/prototype/transforms/_auto_augment.py

```diff
@@ -0,0 +1,417 @@
+            )
+        )
+        image_size = F.get_image_size(image)
+
+        for transform_id, magnitude in self._get_transforms_meta(image_size):
+            dispatcher = self._DISPATCHER_MAP[transform_id]
+
+            def transform(input: Any) -> Any:
+                if not self._is_supported(input):
+                    return input
+
+                return dispatcher(input, magnitude, params["interpolation"], params["fill"])  # type: ignore[index]
+
+            sample = apply_recursively(transform, sample)
+
+        return sample
+
+
+class RandAugment(_AutoAugmentBase):
+    _AUGMENTATION_SPACE = (
+        AugmentationMeta("Identity", lambda num_bins, image_size: None, False),
+        AugmentationMeta("ShearX", lambda num_bins, image_size: torch.linspace(0.0, 0.3, num_bins), True),
+        AugmentationMeta("ShearY", lambda num_bins, image_size: torch.linspace(0.0, 0.3, num_bins), True),
+        AugmentationMeta(
+            "TranslateX",
+            lambda num_bins, image_size: torch.linspace(0.0, 150.0 / 331.0 * image_size[1], num_bins),
+            True,
+        ),
+        AugmentationMeta(
+            "TranslateY",
+            lambda num_bins, image_size: torch.linspace(0.0, 150.0 / 331.0 * image_size[0], num_bins),
+            True,
+        ),
+        AugmentationMeta("Rotate", lambda num_bins, image_size: torch.linspace(0.0, 30.0, num_bins), True),
+        AugmentationMeta("Brightness", lambda num_bins, image_size: torch.linspace(0.0, 0.9, num_bins), True),
+        AugmentationMeta("Color", lambda num_bins, image_size: torch.linspace(0.0, 0.9, num_bins), True),
+        AugmentationMeta("Contrast", lambda num_bins, image_size: torch.linspace(0.0, 0.9, num_bins), True),
+        AugmentationMeta("Sharpness", lambda num_bins, image_size: torch.linspace(0.0, 0.9, num_bins), True),
+        AugmentationMeta(
+            "Posterize",
+            lambda num_bins, image_size: cast(torch.Tensor, 8 - (torch.arange(num_bins) / ((num_bins - 1) / 4)))
+            .round()
+            .int(),
+            False,
+        ),
+        AugmentationMeta("Solarize", lambda num_bins, image_size: torch.linspace(255.0, 0.0, num_bins), False),
+        AugmentationMeta("AutoContrast", lambda num_bins, image_size: None, False),
+        AugmentationMeta("Equalize", lambda num_bins, image_size: None, False),
+    )
+
+    def __init__(self, *, num_ops: int = 2, magnitude: int = 9, num_magnitude_bins: int = 31, **kwargs: Any) -> None:
+        super().__init__(**kwargs)
+        self.num_ops = num_ops
+        self.magnitude = magnitude
+        self.num_magnitude_bins = num_magnitude_bins
+
+    def _get_transforms_meta(self, image_size: Tuple[int, int]) -> Iterator[Tuple[str, float]]:
+        for _ in range(self.num_ops):
+            augmentation_meta = self._AUGMENTATION_SPACE[int(torch.randint(len(self._AUGMENTATION_SPACE), ()))]
+            if augmentation_meta.dispatcher_id == "Identity":
```

Comment:
Nit: Add an "Identity" transform at `_DISPATCHER_MAP` of Base to avoid these if statements. Having Identity is very common on space and it will keep the keys between `_DISPATCHER_MAP` and `_AUGMENTATION_SPACE` synced.